proto/storage/process_listening_on_port.proto

```diff
@@ -57,4 +58,8 @@ message ProcessListeningOnPortStorage {
     // ProcessIndicator will be not empty only for those cases when we were not
     // able to find references process in the database
     ProcessIndicatorUniqueKey process = 7;
+
+    //  string deployment_id = 8 [(gogoproto.moretags) = 'search:"Deployment ID,store,hidden" policy:",prefer-parent" sql:"index=hash,fk(Deployment:id),no-fk-constraint,type(uuid)"'];
+    //string deployment_id = 8 [(gogoproto.moretags) = 'search:"Deployment ID,store" sql:"index=hash,type(uuid)"'];
+    string deployment_id = 8 [(gogoproto.moretags) = 'search:"Deployment ID,store" sql:"type(uuid)"'];
```

Comment:
I am not confidant about this.

Reply:
hash index might make sense on that column

Reply:
I'm afraid this is not immediately obvious. Hash index maybe good for highly unique values, but here `deployment_id` will most likely be quite repeating. The docs [[1]] are saying:

> As a result of the overflow cases, we can say that hash indexes are most suitable for unique, nearly unique data or data with a low number of rows per hash bucket. One possible way to avoid problems is to exclude highly non-unique values from the index using a partial index condition, but this may not be suitable in many cases.

As a small and dirty experiment I've created two tables with only one uuid column, one table having a hash index, another a btree. Then retail inserting 300k records with only 3 distinct values (so, a lot of repetition) shows this:

```
-- insert into the table with btree index
=# insert into test_index_btree select
   case
     when i % 3 = 0 then '1b7e7c08-4c69-4189-9e7d-7474c872f973'::uuid
     when i % 3 = 1 then '98034214-0ad9-41da-a6f8-fefe37224c7b'::uuid
     when i % 3 = 2 then '2f75ed5a-fdbc-454a-8a0b-d22c441583d0'::uuid
   end
   from generate_series(1, 300000) i;

INSERT 0 300000
Time: 1722.175 ms (00:01.722)

-- insert into the table with hash index, somehow slower
=# insert into test_index_hash select
   case
     when i % 3 = 0 then '1b7e7c08-4c69-4189-9e7d-7474c872f973'::uuid
	 when i % 3 = 1 then '98034214-0ad9-41da-a6f8-fefe37224c7b'::uuid
	 when i % 3 = 2 then '2f75ed5a-fdbc-454a-8a0b-d22c441583d0'::uuid
   end
   from generate_series(1, 300000) i;

INSERT 0 300000
Time: 14444.927 ms (00:14.445)

-- size of the involved relations and indexes,
-- hash is almost the same size as the table,
-- where btree is smaller.

             relation             |    size
----------------------------------+------------
 public.test_index_hash_data_idx  | 14 MB
 public.test_index                | 13 MB
 public.test_index_btree          | 13 MB
 public.test_index_btree_data_idx | 1928 kB

```

I'm not saying it's a definitive proof, but probably makes sense to go with btree first, while doing perf tests to compare with hash on the real data distributions. 

[1]: https://www.postgresql.org/docs/current/hash-intro.html

Reply:
sounds reasonable